### ISSUE TITLE
Some more possible destinations

### ISSUE THREAD

**Issue by mrienstra** [kachkaev/njt#174]
Neat tool! I can see how it will save time if I remember to use it!

Some of these are kinda random, some of these I actually use, but thought I'd just go for it, in case any of them strike a chord.

https://npmtrends.com/njt

`npm-audit`: https://www.gyanblog.com/tutorials/how-node-npm-audit-rest-api-vulnerability/

https://npmgraph.js.org/?q=njt

https://snyk.io/advisor/npm-package/njt

https://socket.dev/npm/package/njt

https://techgaun.github.io/active-forks/index.html#https://github.com/kachkaev/njt

**Comment by kachkaev** [comment]
Thanks for sharing these links @mrienstra – some of them are new to me!

I'd be interested in adding more tools, but I’m not quite sure how to elegantly go beyond one-character keywords. I’m afraid that if we introduce too many shortcuts, people may start jumping to wrong URLs, which will bring more harm than benefit.

I wonder if there’s room for something like `d` for _dashboard_, which would open `njt.vercel.app/-/packge-name`. That new page can contain links to all tools including those without direct shortcuts 🤔 WDYT?

**Comment by mrienstra** [comment]
That's a cool idea! Although I think I'd prefer keeping that second step in the CLI, something like this:

1. User enters `njt <package>` without a destination (current behavior would be to default to `n`)
2. Show a list of destinations sorted by keyword, something like this:
<img width="435" alt="image" src="https://user-images.githubusercontent.com/669326/193679927-dd642e8d-1d57-4c37-baea-067195fb417d.png">

3. ... with a prompt as well of course. Accepting input as soon as it matches a keyword (and if you introduce two-character keywords, if there isn't a longer keyword that start with the same character) without needing to hit enter would help keep the number of keystrokes low.
4. A nice touch would be to delete most of those lines from the CLI after a choice was made, leaving only a summary of the flow.

Here's how that might look with 2-character keywords:
<img width="440" alt="image" src="https://user-images.githubusercontent.com/669326/193682174-f9a2f69f-b560-4818-bee9-f106c927af08.png">
(added `pp` from https://github.com/kachkaev/njt/issues/172)

Personally, I don't think two-character keywords would be all that terrible, but I definitely think most users will only be a actively using a few.

I've been pondering the addition of a config, so users can change the default keywords as desired, maybe disable destinations they don't plan to use, or otherwise tweak the behavior.

**Comment by mrienstra** [comment]
If you decide to go this route, check out [`prompts`](https://github.com/terkelg/prompts). I ran across it recently while contributing to `astro`, which uses `prompts` for [two CLI tools](https://github.com/withastro/astro/search?q=%22import+prompts%22).

**Comment by mrienstra** [comment]
Down to work on this if you'd like.

Also, small detail, but the prompt could show a default entry of `n` for npmjs.com, so immediately pressing enter (without other entry) would take you to npmjs.com.

**Comment by kachkaev** [comment]
👋 @mrienstra! `njt` CLI is just one of several user interfaces (entry points) for `njt`. If we add more keywords and figure out UX for `cli`, other environments like browsers, Alfred, etc. may degrade UX-wise. More keywords means more cognitive load on a human and therefore higher chances of landing on the wrong page.

Before we add more keywords, what we probably need is a universal way of providing search prompts for the query (like Google does when we type text in the browser’s address bar). Maybe we should explore this direction first, find a universal solution and then port it to all interfaces we have (which will include CLI). WDYT?

**Comment by iamandrewluca** [comment]
Some of my insights.

I'm using this package extensively, daily.

My most used shortcuts are `s` `n` `b` `u` `c` `d`

I'm using this package because it increases speed drastically.
I even had an attempt to add a cache layer for `njt` results to increase the speed even more.

Personally, I wouldn't mind having shortcuts that will use 2 letters. After you learn them it is very easy to move around.

Having a way to set your own, or disable some shortcuts could also be a nice solution.

**Comment by kachkaev** [comment]
I suggest that we figure out search autocompletion first as it will help with picking the right destination.

> As a user, I should be able to type `njt[space]prettier[space]` and see a list of destinations in most njt UIs.

I don't have any experience with this kind of stuff bu I believe that we need a new API endpoint which can be auto-discovered by Chrome, FF, Alfred etc. When this is sorted, we can add autocompletion support to https://njt.vercel.app and the CLI. This will unblock pretty much any number of destinations we want. Adding two-letter destinations now will make the tool harder to use because it's difficult to remember a lot of abbreviations.

**Comment by iamandrewluca** [comment]
Started reading these two pages, maybe is something useful 🤔 

- https://developers.google.com/custom-search/docs/element
- https://developers.google.com/custom-search/v1/overview

**Comment by iamandrewluca** [comment]
One more possible destination

> Find the true size of an npm package.
> https://pkg-size.dev

**Comment by iamandrewluca** [comment]
One more possible destination

> browse type documentation for npm packages
> https://tsdocs.dev

**Comment by iamandrewluca** [comment]
> a quick npm package size checker
> https://bundlejs.com/

**Comment by iamandrewluca** [comment]
> a quick npm package size checker 
> https://bundlejs.com

**Comment by unrevised6419** [comment]
> Visualize the dependency tree of a package or project to see where you might want to optimize
> https://deptree.rschristian.dev/

**Comment by kachkaev** [comment]
Hey folks! Just added `x` for https://npmx.dev/ in https://github.com/kachkaev/njt/pull/1133. A new version of CLI is also [released](https://github.com/kachkaev/njt/releases).

Two-letter destinations are still blocked – haven't found time to look into https://github.com/kachkaev/njt/issues/174#issuecomment-1325266791 yet. But if you still see destination ideas, let's keep listing them here. Cheers!

**Comment by kachkaev** [comment]
It is possible to add collect stats for package names and destinations in [jump.handler.ts](https://github.com/kachkaev/njt/blob/d82c9e1ec31b527ce268063d3a14f1ee40bf1b92/pages/api/jump.handler.ts). I haven't done it yet, but maybe it's more feasible with coding agents than before. Happy to discuss this in a separate issue if you're interested with helping.

PS: Using njt via browser each day as well.
